boa/src/builtins/array/mod.rs

```diff
@@ -1782,6 +1783,188 @@ impl Array {
+        // 6. Else, let actualStart be min(relativeStart, len).
+        let actual_start = Self::get_relative_start(context, args.get(0), len)?;
+        // 7. If start is not present, then
+        let insert_count = if args.get(0).is_none() || args.get(1).is_none() {
+            // 7a. Let insertCount be 0.
+            // 8. Else if deleteCount is not present, then
+            // a. Let insertCount be 0.
+            0
+        // 9. Else,
+        } else {
+            // 9a. Let insertCount be the number of elements in items.
+            args.len().saturating_sub(2)
+        };
+        let actual_delete_count = if args.get(0).is_none() {
+            // 7b. Let actualDeleteCount be 0.
+            0
+            // 8. Else if deleteCount is not present, then
+        } else if args.get(1).is_none() {
+            // 8b. Let actualDeleteCount be len - actualStart.
+            len - actual_start
+        // 9. Else,
+        } else {
+            // b. Let dc be ? ToIntegerOrInfinity(deleteCount).
+            let dc = args
+                .get(1)
+                .cloned()
+                .unwrap_or_default()
+                .to_integer_or_infinity(context)?;
+            // c. Let actualDeleteCount be the result of clamping dc between 0 and len - actualStart.
+            let max = len - actual_start;
+            match dc {
+                IntegerOrInfinity::Integer(i) => {
+                    if i < 0 {
+                        0
+                    } else if i as usize > max {
+                        max
+                    } else {
+                        i as usize
+                    }
+                }
+                IntegerOrInfinity::PositiveInfinity => max,
+                IntegerOrInfinity::NegativeInfinity => 0,
+            }
+        };
+
+        // 10. If len + insertCount - actualDeleteCount > 253 - 1, throw a TypeError exception.
+        if len + insert_count - actual_delete_count > Number::MAX_SAFE_INTEGER as usize {
+            return context.throw_type_error("Target splice exceeded max safe integer value");
+        }
+
+        // 11. Let A be ? ArraySpeciesCreate(O, actualDeleteCount).
+        let arr = Self::array_species_create(&o, actual_delete_count, context)?;
+        // 12. Let k be 0.
+        // 13. Repeat, while k < actualDeleteCount,
+        for k in 0..actual_delete_count {
+            // a. Let from be ! ToString(𝔽(actualStart + k)).
+            // b. Let fromPresent be ? HasProperty(O, from).
+            let from_present = o.has_property(actual_start + k, context)?;
+            // c. If fromPresent is true, then
+            if from_present {
+                // i. Let fromValue be ? Get(O, from).
+                let from_value = o.get(actual_start + k, context)?;
+                // ii. Perform ? CreateDataPropertyOrThrow(A, ! ToString(𝔽(k)), fromValue).
+                arr.create_data_property_or_throw(actual_start + k, from_value, context)?;
```

Comment:
```suggestion
                arr.create_data_property_or_throw(k, from_value, context)?;
```
The spec creates the `k` property, not the `actual_start + k` property.

Reply:
Sorry, I completely missed this! Good spot.